templates/swift/Sources/Query.swift.twig

```diff
@@ -1,104 +1,272 @@
+        }
     }
+}
+public struct Query : Encodable, CustomStringConvertible {
+    var method: String
+    var attribute: String?
+    var values: [QueryValue]?
 
-    public static func notEqual(_ attribute: String, value: Any) -> String {
-        buildQueryWhere(attribute, is: "notEqual", to: value)
+    init(method: String, attribute: String? = nil, values: Any? = nil) {
+        self.method = method
+        self.attribute = attribute
+        self.values = Query.convertToQueryValueArray(values)
     }
 
-    public static func lessThan(_ attribute: String, value: Any) -> String {
-        buildQueryWhere(attribute, is: "lessThan", to: value)
+    private static func convertToQueryValueArray(_ values: Any?) -> [QueryValue]? {
+        if (values is [QueryValue] || values == nil) {
+            return values as? [QueryValue]
+        } 
+
+        if (values is [String]) {
+            return (values as! [String]).map { QueryValue.string($0) }
+        }
+
+        if (values is [Int]) {
+            return (values as! [Int]).map { QueryValue.int($0) }
+        }
+
+        if (values is [Double]) {
+            return (values as! [Double]).map { QueryValue.double($0) }
+        }
+
+        if (values is [Bool]) {
+            return (values as! [Bool]).map { QueryValue.bool($0) }
+        }
+
+        if (values is [Query]) {
+            return (values as! [Query]).map { QueryValue.query($0) }
+        }
+
+        return nil
     }
 
-    public static func lessThanEqual(attribute: String, value: Any) -> String {
-        buildQueryWhere(attribute, is: "lessThanEqual", to: value)
+    enum CodingKeys: String, CodingKey {
+        case method
+        case attribute
+        case values
     }
 
-    public static func greaterThan(_ attribute: String, value: Any) -> String {
-        buildQueryWhere(attribute, is: "greaterThan", to: value)
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(method, forKey: .method)
+        if (self.attribute != nil) {
+            try container.encode(attribute, forKey: .attribute)
+        }
+        if (values != nil) {
+            try container.encode(values, forKey: .values)
+        }
     }
 
-    public static func greaterThanEqual(_ attribute: String, value: Any) -> String {
-        buildQueryWhere(attribute, is: "greaterThanEqual", to: value)
+    public var description: String {
+        do {
+            let data = try JSONEncoder().encode(self)
+            return String(data: data, encoding: .utf8) ?? ""
+        } catch {
+            return ""
+        }
+
     }
 
-    public static func isNull(_ attribute: String) -> String {
-        "isNull(\"\(attribute)\")"
+    public static func equal(_ attribute: String, value: Any) -> Query {
+        return Query(
+            method: "equal",
+            attribute: attribute,
+            values: [value]
```

Comment:
We need to make sure `value` isn't already an array for methods that accept `Any`